modules/table/src/main/java/org/apache/ignite/internal/utils/RebalanceUtil.java

```diff
@@ -202,13 +222,46 @@ public static ByteArray stablePartAssignmentsKey(String partId) {
     }
 
     /**
-     * Extract table id from pending key of partition.
+     * Key that is needed for the rebalance algorithm.
+     *
+     * @param partId Unique identifier of a partition.
+     * @return Key for a partition.
+     * @see <a href="https://github.com/apache/ignite-3/blob/main/modules/table/tech-notes/rebalance.md">Rebalnce documentation</a>
+     */
+    public static ByteArray switchReduceKey(String partId) {
+        return new ByteArray(ASSIGNMENTS_SWITCH_REDUCE_PREFIX + partId);
+    }
+
+    /**
+     * Key that is needed for the rebalance algorithm.
+     *
+     * @param partId Unique identifier of a partition.
+     * @return Key for a partition.
+     * @see <a href="https://github.com/apache/ignite-3/blob/main/modules/table/tech-notes/rebalance.md">Rebalnce documentation</a>
+     */
+    public static ByteArray switchAppendKey(String partId) {
+        return new ByteArray(ASSIGNMENTS_SWITCH_APPEND_PREFIX + partId);
+    }
+
+    /**
+     * Extract table id from a metastorage key of partition.
+     *
+     * @param key Key.
+     * @return Table id.
+     */
+    public static UUID extractTableId(ByteArray key) {
+        return extractTableId(key, "");
+    }
+
+    /**
+     * Extract table id from a metastorage key of partition.
      *
      * @param key Key.
+     * @param prefix Key prefix.
      * @return Table id.
      */
     public static UUID extractTableId(ByteArray key, String prefix) {
-        var strKey = key.toString();
+        String strKey = key.toString();
 
         return UUID.fromString(strKey.substring(prefix.length(), strKey.indexOf("_part_")));
```

Comment:
Again, it has nothing to do with your code, but don't UUIDs have a fixed length? What the hell is going on with these classes

Reply:
They do have fixed length, but what do you mean?

Reply:
I mean that calling `indexOf` makes no sense! It can be derived from `prefix.length()`. Author clearly didn't have time to think about performance

Reply:
I don't like this method at all, extracting table id from the key seems odd

Reply:
Remember when metastorage was in the design phase and people were thinking "we can store keys effectively, like storing integers as 4 bytes instead of their decimal representation...". Everyone forgot about it. But I digress.
Storing tableId in keys is absolutely fine